src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -1116,25 +1116,36 @@ let doc_val ctx pat exp =
   let base_pp = doc_exp false ctx exp in
   nest 2 (group (string "def" ^^ space ^^ idpp ^^ typpp ^^ space ^^ coloneq ^/^ base_pp))
 
-let rec doc_defs_rec ctx defs types docdefs =
+let def_prints_function def =
```

Comment:
Maybe use a name that makes it more clear it returns a boolean, like `should_print_function_def`?